src/core/utils.jl

```diff
@@ -613,4 +613,271 @@ function error_if_series_vals_not_0_to_1(series, input_struct_name, input_name)
+    else
+        if "Financial" in keys(defaults)
+            for input_key in ["elec_cost_escalation_rate_fraction", "existing_boiler_fuel_cost_escalation_rate_fraction",
+                                "boiler_fuel_cost_escalation_rate_fraction", "chp_fuel_cost_escalation_rate_fraction",
+                                "generator_fuel_cost_escalation_rate_fraction"]
+                defaults["Financial"][input_key] = defaults["Financial"][input_key][federal_escalation_region]
+            end
+        else
+            for key in keys(defaults)
+                filter_sector_defaults_by_region!(defaults[key]; federal_escalation_region=federal_escalation_region)
+            end
+        end
+    end
+end
+function get_all_sector_defaults()
+    sector_defaults_path = joinpath(@__DIR__, "..", "..", "data", "sector_dependent_defaults.json")
+    if !isfile(sector_defaults_path)
+        throw(ErrorException("sector_dependent_defaults.json not found at path: $sector_defaults_path"))
+    end
+    sector_defaults = JSON.parsefile(sector_defaults_path)
+    return sector_defaults
+end
+function get_sector_defaults(; sector::String, federal_procurement_type::String="", federal_sector_state::String="", struct_name::String="")
+    sector_defaults = get_all_sector_defaults()
+    
+    if sector=="federal"
+        if isempty(federal_procurement_type) 
+            throw(@error("federal_procurement_type must be provided to get_sector_defaults() when sector is 'federal'"))
+        end
+        sector_defaults = get(get(sector_defaults, sector, Dict{String,Any}()), federal_procurement_type, Dict{String,Any}())
+        federal_escalation_region = get_NIST_EERC_rate_region(federal_sector_state)
+        if struct_name == "Financial" || isempty(struct_name)
+            if isempty(federal_escalation_region)
+                @warn "No or invalid federal_sector_state provided, so national average used for default federal escalation rates."
+                federal_escalation_region = "National"
+            end
+            filter_sector_defaults_by_region!(sector_defaults; federal_escalation_region=federal_escalation_region)
+        end
+    else
+        sector_defaults = get(sector_defaults, sector, Dict{String,Any}())
+    end
+
+    if !isempty(struct_name)
+        sector_defaults = get(sector_defaults, struct_name, Dict{String,Any}())
+    end
+    
+    return sector_defaults
+end
+function set_sector_defaults!(d::Dict; struct_name::String, sector::String, federal_procurement_type::String="", federal_sector_state::String="")
+    sector_defaults = get_sector_defaults(; sector=sector, federal_procurement_type=federal_procurement_type, federal_sector_state=federal_sector_state, struct_name=struct_name)
+    for (input_name, input_val) in sector_defaults
+        if !(input_name in keys(d))
+            d[input_name] = input_val
+        end
+    end
+end
+
+# function get_sector_defaults_techs(; sector::String, federal_procurement_type::String)
```

Comment:
@hdunham  can this commented out code be removed?

Reply:
oops yes I thought I removed this but it must have been in the other 15 branches I created while debugging!